lib/Authentication/AuthTokens/ShibAuthToken.php

```diff
@@ -96,73 +97,67 @@ private function getAttributesInitToken(){
-                $HTML .= "<div style='text-align: center;'>";
-                $HTML .= '<a href="'.htmlspecialchars(\Factory::$properties['LOGOUTURL']).'"><b><font colour="red">Logout</font></b></a>';
-                $HTML .= "</div>";
-                echo ($HTML);
-                die();
-            }
 
-            $entitlementValuesArray = explode(';', $_SERVER['entitlement']);
-            if( !in_array('urn:mace:egi.eu:res:gocdb#aai.egi.eu', $entitlementValuesArray) ){
-                 $HTML = '<ul><li>Login requires a GOCDB entitlement <a href="https://wiki.egi.eu/wiki/URN_Registry:aai.egi.eu:gocdb" target="_blank">https://wiki.egi.eu/wiki/URN_Registry:aai.egi.eu:gocdb</a></li><li>Please, logout or restart your browser and attempt to login again using an identity provider that provides a GOCDB entitlement</li></ul>';
-                 $HTML .= "<div style='text-align: center;'>";
-                 $HTML .= '<a href="'.htmlspecialchars(\Factory::$properties['LOGOUTURL']).'"><b><font colour="red">Logout</font></b></a>';
-                 $HTML .= "</div>";
-                 echo ($HTML);
-                 die();
+        $configService = \Factory::getConfigService();
+        $identityProviders = $configService->getIdentityProvidersInfo();
+
+        foreach ($identityProviders as $provider) {
+            if ($provider['idp'] === $idp) {
+                $name = $provider['name'];
+                $helpUrl = $provider['help_url'];
+
+                if (empty($_SERVER['voPersonID'])) {
+                    die(
+                        "Did not receive required attributes from the "
+                        . "IDP $name to complete authentication. "
+                        . "Please contact gocdb-admins."
+                    );
+                }
+
+                if (empty($_SERVER['entitlement'])) {
+                    die(
+                        "Did not receive the required entitlement "
+                        . "attribute from the IDP $name. "
+                        . "Please contact gocdb-admins."
+                    );
+                }
+
+                if (!empty($provider['required_groups'])) {
+                    $entitlementValues = explode(';', $_SERVER['entitlement']);
+
+                    if (
+                        !array_intersect(
+                            $entitlementValues,
+                            $provider['required_groups']
+                        )
+                    ) {
+                        $HTML = "<ul>"
+                            . "<li>Login requires a GOCDB entitlement value "
+                            . "which was not provided for the IDP $name.</li>"
```

Comment:
I think this will read better

```suggestion
                            . "which was not provided for the $name.</li>"
```